### PR TITLE
Pref: Remove Unit tab and move settings back to main page.

### PR DESCRIPTION
**Pull request by PaddleStroke**
This has been discussed on the forum and everyone seems to agree about it.
Also the main page just lost 2 lines since the wb pref PR merged. So now it's not over-crowded.

Description : 
- Moves content of Unit tab to General tab.
- Fraction of inch is invisible if the unit system is not US building instead of just deactivated.
- Number of decimal is on the same line as unit system, so the whole unit page fit on one line.
- Change the groupbox name from "Language" to "Language and number format".
- Change the QLabel "Change language" to "Language"

Issue : https://github.com/FreeCAD/FreeCAD/issues/9137

Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [ ]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

File: src/Gui/DlgGeneral.ui [hunk 1]
```diff
@@ -71,13 +71,100 @@
...
+          <widget class="QLabel" name="unitSystemLabel">
+           <property name="text">
+            <string>Unit system:</string>
+           </property>
+          </widget>
+         </item>
+         <item row="1" column="1">
+          <widget class="QComboBox" name="comboBox_UnitSystem">
+           <property name="toolTip">
+            <string>Unit system that should be used for all parts the application</string>
```
Comment (chennes) [inline src/Gui/DlgGeneral.ui:83]: "for all parts *of* the application"

File: src/Gui/DlgGeneralImp.cpp [hunk 2]
```diff
@@ -462,4 +536,23 @@ void DlgGeneralImp::onLoadPreferencePackClicked(const std::string& packName)
     }
 }
 
+
+void DlgGeneralImp::on_comboBox_UnitSystem_currentIndexChanged(int index)
```
Comment (chennes) [inline src/Gui/DlgGeneralImp.cpp:540]: This style of function name makes it look like an auto-connected slot. If it is, it should not be: Clazy will complain about it, that use is no longer recommended. (Sorry, on a phone, it's hard to see for sure).

**Review by chennes — CHANGES_REQUESTED** [review]
Thanks for the PR: couple of minor things.

File: src/Gui/DlgGeneralImp.cpp (hunk 2 above)
Comment (PaddleStroke) [reply src/Gui/DlgGeneralImp.cpp:540]: In this PR I only copy pasted without having a more indepth look.
It appears that it is indeed auto-connected. I can change that.

**Comment by chennes** [comment]
I haven't checked, but I assume this will require some updates on the Wiki.

**Comment by PaddleStroke** [comment]
I guess so. However there might be further changes to the main page, so it
may not be useful to change now.

**Pull request merged**